File: examples/node/batch-process-folder.ts
import { mkdirSync, readdirSync, readFileSync, writeFileSync } from "node:fs";
import { join } from "node:path";
import { AILabClient } from "ailabtools";

const client = new AILabClient({ apiKey: process.env.AILAB_API_KEY! });

async function main() {
  const inputDir = "./input";
  const outputDir = "./output";
  mkdirSync(outputDir, { recursive: true });

  for (const name of readdirSync(inputDir)) {
    const result = await client.background.remove({
      image: readFileSync(join(inputDir, name)),
      returnForm: "whiteBK",
    });

    writeFileSync(join(outputDir, `${name}.json`), JSON.stringify(result, null, 2));
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: examples/python/batch_process_folder.py
import asyncio
import json
import os
from pathlib import Path
from ailabtools import AILabClient


async def main():
    client = AILabClient(api_key=os.environ["AILAB_API_KEY"])
    input_dir = Path("./input")
    output_dir = Path("./output")
    output_dir.mkdir(exist_ok=True)

    for path in input_dir.iterdir():
        if not path.is_file():
            continue
        with path.open("rb") as image:
            result = await client.background.remove({"image": image, "returnForm": "whiteBK"})
        (output_dir / f"{path.name}.json").write_text(json.dumps(result, indent=2))

    await client.aclose()


asyncio.run(main())
